src/commands/logStream/RevisionListStep.ts

```diff
@@ -13,15 +13,18 @@ import { IStreamLogsContext } from "./IStreamLogsContext";
 
 export class RevisionListStep extends AzureWizardPromptStep<IStreamLogsContext> {
     public async prompt(context: IStreamLogsContext): Promise<void> {
-        if (context.containerApp.revisionsMode === 'Multiple') {
-            const placeHolder: string = localize('selectRevision', 'Select a revision');
-            context.revision = (await context.ui.showQuickPick(this.getPicks(context), { placeHolder })).data;
-        } else {
-            (await this.getPicks(context)).forEach(revision => {
-                if (revision.data.active === true) {
-                    context.revision = revision.data;
-                }
-            });
+        const placeHolder: string = localize('selectRevision', 'Select a revision');
+        context.revision = (await context.ui.showQuickPick(this.getPicks(context), { placeHolder })).data;
+    }
+
+    public async configureBeforePrompt(context: IStreamLogsContext): Promise<void> {
+        const picks = await this.getPicks(context);
```

Comment:
We don't want to call this first. In the event that this is single revision mode, this call was pointless and makes the wizard slightly more sluggish.

I'd suggestion that you first check if the revisionsMode is single, then in your else clause, do the logic to see how many revisions there are.

Reply:
You may also want to cache the results of `this.getPicks` in the wizard step, otherwise we will call it again needlessly if the event we actually do prompt the users.

Since it's pretty rare to actually have multiple revision mode, I wouldn't mind too much if you don't cache it, but I thought I'd mention it for best practice :)